### PR TITLE
Replace Controller.Json() by ControllerBase.BadRequest() to return a 400 response for invalid token requests

### PR DESCRIPTION
**Pull request by kevinchalet**
Fixes https://github.com/openiddict/openiddict-core/issues/209.

@jayrulez care to review? :smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,14 +156,16 @@ To **support the password or the client credentials flow, you must provide your
 
 ```csharp
 [HttpPost("~/connect/token")]
+[Produces("application/json")]
```
Comment (jayrulez) [inline README.md:159]: Is this line necessary?

I get application/json content type in the response headers when i sent that header.

**Comment by jayrulez** [comment]
Looks good @PinpointTownes

**Comment by jayrulez** [comment]
Ah, I didn't see your comment on the issue before I reviewed. Since we want to ensure that the json formatter serializes the object then disregard my question above.

File: README.md (hunk 1 above)
Comment (kevinchalet) [reply README.md:159]: If you have multiple formatters (e.g if you register the XML formatter), I think there's a risk of getting XML if the client sent an `Accept` with XML specified before JSON.

Comment (jayrulez) [reply README.md:159]: I never really thought about that before. Thanks for pointing that out.

**Comment by kevinchalet** [comment]
Thanks for the review :+1:

**Pull request merged**